src/main/java/net/minecraftforge/server/permission/nodes/PermissionNode.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission.nodes;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+
+/**
+ * @see PermissionNode#PermissionNode(String, PermissionType, PermissionResolver, PermissionDynamic[])
+ * @param <T> PermissionType see {@link PermissionTypes}
+ */
+public class PermissionNode<T>
```

Comment:
```
Represents the basic unit at the heart of the permission system.

<p>A permission indicates the ability for an actor to perform an action, in its most general sense. In the permission system, all permissions are encoded as instances of this class, optionally integrated by a {@link PermissionDynamicContext}.</p>

<p>A node is uniquely identified by its `nodeName`, which is a dot-separated string providing meaning to the node itself. The suggested structure of the name is `modid.name`, where `modid` is the ID of the owner of the node. This API <strong>does not require</strong> any implicit hierarchy, so `modid.name` is not considered a parent of `modid.name.sub`. Such decisions are left to the {@link IPermissionHandler}.</p>

<p>Each node also has an associated {@link Permission Type}, detailing its exact type, along with a {@link PermissionResolver} that is used to obtain the default value of the permission. More information can be found on their documentation.</p>

<p>A node can also be bound to additional {@link PermissionDynamic}s, which are used when querying the availability of the permission. There are no restrictions on their amount. It is also not mandatory to provide a value for all dynamics in a permission query. See the above link for more information.</p>

<p>Each node should be registered via the {@link Event} and stored statically in a field. That instance should then be reused every-time a permission check needs to be performed via {@link PermissionAPI#check}.</p>
```

Not the best, but I feel this describes more coherently the class and gives more information for someone that is just getting started. A good idea would be to reference nodes in the documentation for `PermissionAPI` too, as to give a clearer connection.

Reply:
You, are a master at writing Java Docs :eyes: Love it!